### PR TITLE
Update CSRMatrix and push coverage to 100%, add to_uint_array util

### PR DESCRIPTION
**Pull request by rly**
## Motivation

With the change in hdmf-common-schema of dtypes in `CSRMatrix` from int to uint (https://github.com/hdmf-dev/hdmf-common-schema/pull/46), the API needs to be updated in order to avoid data type conversion warnings.

Also add testing of the different error and edge cases of `CSRMatrix`.

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/hdmf-dev/hdmf/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

**Review by ajtritt — DISMISSED** [review]

File: src/hdmf/common/sparse.py [hunk 1]
```diff
@@ -22,32 +22,37 @@ def __init__(self, **kwargs):
...
+            raise ValueError("Cannot convert '%s' to an array of unsigned integers." % arg) from ve
+        if ar.ndim != 1:
+            raise ValueError("'%s' must be a 1D array of unsigned integers." % arg)
+        return ar
 
     def __getattr__(self, val):
-        return getattr(self.__data, val)
-
-    @property
-    def shape(self):
```
Comment (oruebel) [inline src/hdmf/common/sparse.py:49]: Is there a particular reason for removing the ``shape`` property?

**Review by oruebel — DISMISSED** [review]

File: src/hdmf/common/sparse.py (hunk 1 above)
Comment (rly) [reply src/hdmf/common/sparse.py:49]: It is redundant with `self.__getattr__('shape')` which defers to `self.__data.shape` (equal to before), but then the values are converted to uints.

**Review by ajtritt — APPROVED** [review]

File: src/hdmf/common/sparse.py (hunk 1 above)
Comment (oruebel) [reply src/hdmf/common/sparse.py:49]: Thanks for the clarification. That makes sense.

**Review by oruebel — APPROVED** [review]
